### PR TITLE
RMC-221 CCS case look up by postcode 

### PR DESCRIPTION
**Pull request by AdamHawtin**
# Motivation and Context
Contact centre needs to be able to search for CCS cases by postcode. 

# What has changed
* Add find CCS cases by postcode endpoint
* Tests for new endpoint

# How to test?
Build and run this image with docker dev. Load some CCS cases, you should be able to hit this endpoint with their postcode and get them back.
Run AT's on the linked branch.

# Links
https://trello.com/c/RxoE4p0V/522-rmc-221-provide-ccs-case-look-up-by-post-code-8
https://github.com/ONSdigital/census-rm-acceptance-tests/pull/168

### CODE REVIEW

**Comment by AdamHawtin** [comment]
[RMC-221 Provide CCS case look up by post code (8)](https://trello.com/c/RxoE4p0V/522-rmc-221-provide-ccs-case-look-up-by-post-code-8)

File: src/main/java/uk/gov/ons/census/caseapisvc/model/repository/CaseRepository.java [hunk 1]
```diff
@@ -13,5 +15,9 @@
 
   Optional<Case> findByCaseRef(int reference);
 
+  @Query(
+      "SELECT c FROM Case c WHERE survey='CCS' AND UPPER(REPLACE(postcode, ' ', '')) LIKE UPPER(REPLACE(:postcode, ' ', ''))")
```
Comment (NickGrantX) [inline src/main/java/uk/gov/ons/census/caseapisvc/model/repository/CaseRepository.java]: why does it need to be a `LIKE` if you're not using the `%` wildcard? You could just use a regular `=`

**Review by NickGrantX — CHANGES_REQUESTED** [review]
At the moment we're allowing partial postcode searches because you've used `LIKE`. This might be by design, but in theory, somebody could retrieve 400,000 results just by searching for `%`

**Review by LukeLoze1 — APPROVED** [review]

**Comment by AdamHawtin** [comment]
@NickGrantONS Yes, it doesn't need to be a `LIKE`, I've changed it to an equals comparison.

**Review by NickGrantX — APPROVED** [review]
Looks good. Performance testing probably not important because this'll be called relatively infrequently.

File: src/main/java/uk/gov/ons/census/caseapisvc/model/repository/CaseRepository.java (hunk 1 above)
Comment (NickGrantX) [reply src/main/java/uk/gov/ons/census/caseapisvc/model/repository/CaseRepository.java]: Actually you still need to address this review comment

**Review by NickGrantX — CHANGES_REQUESTED** [review]
Still need to replace that `LIKE` with an `=`

**Review by NickGrantX — APPROVED** [review]
Looks good 👍

**Pull request merged**